### ISSUE TITLE
Accessibility regarding PKHex

### ISSUE THREAD

**Issue by Gamulation** [kwsch/PKHeX#3758]
Hi. I just made an issue about keyboard navigation. Just found out it has been done and I am super thankful. I built that one and checked it out. I found a few issues. I am unable to do anything about the pokemon due to the fact that my screen reader don't speak. I only use keyboard and tab navigation. If I go to the box tab and tab more it stops speaking. I know it moves, cause if I press shift tab it goes back to the box tab item thing, but I can't get speech beyond that.
This also happens in the mistery gift area. I am unable to know which slot I put it in. I am now able to put events in the slots which is nice, but I have no idea on what slot.
I really hope this can be looked into. If you are interested in trying this out you can either use the default windows screen reader by holding down the windows key, CTL and pressing enter. Or use this one which I use all the time: https://www.nvaccess.org/download/
To exit the default screen reader just press the same key combo.
Thanks again for making this more keyboard friendly.

**Comment by kwsch** [comment]
I played around with Windows' built in screen Narrator, and after a bit of toil I was able to get it to narrate the picture boxes as I tabbed between them when the focus changed. I also updated some of the other subforms and tabs, in case you ever reach them with enough navigation steps...

Apologies if it is a bit verbose on the descriptions; since Pokémon and mystery gifts have a fair bit of data, there is a lot needed to differentiate them. Hopefully this is a step in the right direction; if there's anything that is unworkable, let me know and I'll see what I can do to accommodate!

**Comment by kwsch** [comment]
When you've selected a Pokémon slot by tabbing into it, try pressing the ContextMenu key on the keyboard. It should bring up the context menu for that slot, which has at least one item to select. Might need to press the down arrow key to set focus to the first item in the list.

**Comment by Gamulation** [comment]
Hi. I don't know if I should make more and more issues, so I just commented on this one to comment on a few things.
1: When changing the level of a pokémon with keyboard it always goes to level 100. When I delete the level and type 20 it still goes to level 100. Could this be fixed?
2: I tried to make a Pokémon shiny, but I'm not sure how to do it. I looked through the tabs, but no idea. Is this just me not finding it or is it something that doesn't read?
Again huge thanks for implementing accessibility in this program!

**Comment by kwsch** [comment]
1. Deleting the text in the textbox would have it reset to 1, with the cursor before the 1. I've removed that behavior, now it just leaves the textbox blank (with the interpreted value still being 1).
2. There's a button with a Star glyph that doesn't annunciate with Narrator, so I added an accessibility description when you tab to it from the PID textbox/Main tab. If it isn't there, then it is currently shiny. The program shows a red star or a yellow square (to denote which shiny it is). I wasn't able to get the program to tab over to those shiny images when shown, so you'll have to rely on the PID textbox's tooltip. The tooltip will indicate the "PSV" and "XOR". When the XOR is 0, it is square shiny. When the XOR is <= 15, it is shiny. (Gen3-5, it is <= 7).

I also revised the gender changing panels so that they'll annunciate when you enter them, and pressing space/enter will try to toggle them to the other state if the species permits.

**Comment by Gamulation** [comment]
Thanks! Now my Staravia can be all shiny and happy.
I have noticed a thing where it says forms. I can't do anything about it, but I'm guessing it's for megas?
I also found another bug. When tabbing it will say new update available. Right before you get to the tab thing to switch between boxws, party and save.
No idea why it says that, as I have latest version.

**Comment by kwsch** [comment]
Due to the layout of the program, there's a hidden label at the top that indicates if an update is available. Usually, it is hidden unless an update is detected. With Narrator, it will still mention it even if it is hidden, and tab stop disabled, so I can't get it to be quiet about it describing the Pokémon editor tabs. I've changed the hidden text so it says "You are using the latest version" instead of New Update Available.

For the forms, there's a custom control that I can't get to be un-selectable while hidden. If the Pokémon has no alternate form argument, that control is hidden from view.

Special Pokémon like Alcremie have base forms, and cosmetic alterations to determine a "sub-form" of sorts. Others like Furfrou use it as a numeric to track days remaining until the form's hairstyle fades back to the default form. And others use it for stat tracking, like Annihilape using Rage Fist.

This value didn't come about until Generation 6 (X/Y), so you'll probably be best off ignoring it. Forms like Unown will have a drop-down to select which A-Z/!/? form they are. Mega forms will also appear in the drop-down, if they are available in the game you are editing.
